src/database.py

```diff
@@ -0,0 +1,215 @@
+logger = logging.getLogger(__name__)
+
+# Define the base for ORM models
+Base = declarative_base()
+
+# ORM model for the 'orders' table
+class Order(Base):
+    __tablename__ = 'orders'
+
+    id = Column(Integer, Sequence('id'), primary_key=True)
+    order_id = Column(String)
+    customer_unique_id = Column(String)
+    order_status = Column(String)
+    order_purchase_timestamp = Column(DateTime)
+    order_approved_at = Column(DateTime)
+    order_delivered_carrier_date = Column(DateTime)
+    order_delivered_customer_date = Column(DateTime)
+    order_estimated_delivery_date = Column(DateTime)
+    item_quantity = Column(Integer)
+    price = Column(Float)
+    freight_value = Column(Float)
+    product_id = Column(String)
+    product_category_name = Column(String)
+    product_category_name_english = Column(String)
+    customer_city = Column(String)
+    seller_id = Column(String)
+    seller_city = Column(String)
+    payment_installments = Column(Integer)
+    payment_value = Column(Float)
+    review_score = Column(Integer)
+
+def setup_database(db_path: str):
+    """Sets up the DuckDB database and returns the engine."""
+    logging.info("Setting up database...")
+    engine = create_engine(f'duckdb:///{db_path}')
+    
+    # Drop existing table if exists
+    inspector = inspect(engine)
+    if 'orders' in inspector.get_table_names():
+        # Drop the existing table
+        Base.metadata.drop_all(tables=[Order.__table__], bind=engine)
+        logging.info("Existing table dropped.")
+
+    Base.metadata.create_all(engine)
+    logging.info("Database setup complete.")
+    return engine
+
+def create_session(engine):
+    """Creates and returns a new database session."""
+    logging.info("Creating a new database session...")
+    Session = sessionmaker(bind=engine)
+    return Session()
+
+def load_csv_data(working_dir: str) -> dict:
+    """Load all necessary CSV files into Polars DataFrames."""
+    logging.info("Loading CSV data into Polars DataFrames...")
+    file_paths = {
+        'customers': 'olist_customers_dataset.csv',
+        'orders': 'olist_orders_dataset.csv',
+        'order_items': 'olist_order_items_dataset.csv',
+        'payments': 'olist_order_payments_dataset.csv',
+        'reviews': 'olist_order_reviews_dataset.csv',
+        'products': 'olist_products_dataset.csv',
+        'sellers': 'olist_sellers_dataset.csv',
+        'product_translation': 'product_category_name_translation.csv'
+    }
+
+    data_frames = {}
+    for key, filename in file_paths.items():
+        full_path = os.path.join(working_dir, filename)
+        if not os.path.exists(full_path):
+            logging.error(f"File not found: {full_path}")
+            raise FileNotFoundError(f"File not found: {full_path}")
+        data_frames[key] = pl.read_csv(full_path)
+        
+    logging.info("Data loaded into DataFrames.")
+    return data_frames
+
+def process_data(data_frames: dict):
```

Comment:
for all functions please add complete type hints for args and responses, across all files, can't be re-commenting this ;)

Reply:
Hi Sogo. This is not very clear to me.